drivers/gpio/gpio_sn74hc595.c

```diff
@@ -26,32 +31,41 @@ LOG_MODULE_REGISTER(gpio_sn74hc595, CONFIG_GPIO_LOG_LEVEL);
 #endif
 
 struct gpio_sn74hc595_config {
-	/* gpio_driver_config needs to be first */
-	struct gpio_driver_config config;
```

Comment:
Sorry, I misguided you regarding `port_pin_mask` :disappointed: 
The field is verified in the `gpio_pin_configure()` syscall here: [include/zephyr/drivers/gpio.h#L1020](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/gpio.h#L1020)
Please return the field back.